src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/MessageBody.cs

```diff
@@ -246,8 +246,8 @@ internal static unsafe void ParseConnection(StringValues connection, out bool is
                                 (*++ch | 0x20) == 'v' &&
                                 (*++ch | 0x20) == 'e')
                             {
-                                isKeepAlive = true;
                                 ch++;
+                                isKeepAlive = ch == end || *ch == ',' || *ch == ' ';
```

Comment:
Is finding a space sufficient? Wouldn't you also need to ensure the space(s) are followed by a ',' or the end of string?